File: app/src/main/java/code/name/monkey/retromusic/ui/fragments/settings/AudioSettings.java
package code.name.monkey.retromusic.ui.fragments.settings;

import android.content.Intent;
import android.content.pm.PackageManager;
import android.content.pm.ResolveInfo;
import android.media.audiofx.AudioEffect;
import android.os.Bundle;
import androidx.preference.Preference;

import code.name.monkey.retromusic.R;
import code.name.monkey.retromusic.util.NavigationUtil;
import code.name.monkey.retromusic.util.PreferenceUtil;

/**
 * @author Hemanth S (h4h13).
 */

public class AudioSettings extends AbsSettingsFragment {
    @Override
    public void invalidateSettings() {
        Preference findPreference = findPreference("equalizer");
        if (!hasEqualizer() && !PreferenceUtil.getInstance().getSelectedEqualizer().equals("retro")) {
            findPreference.setEnabled(false);
            findPreference.setSummary(getResources().getString(R.string.no_equalizer));
        } else {
            findPreference.setEnabled(true);
        }
        findPreference.setOnPreferenceClickListener(preference -> {
            //noinspection ConstantConditions
            NavigationUtil.openEqualizer(getActivity());
            return true;
        });


    }

    private boolean hasEqualizer() {
        final Intent effects = new Intent(AudioEffect.ACTION_DISPLAY_AUDIO_EFFECT_CONTROL_PANEL);
        //noinspection ConstantConditions
        PackageManager pm = getActivity().getPackageManager();
        ResolveInfo ri = pm.resolveActivity(effects, 0);
        return ri != null;
    }

    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
        addPreferencesFromResource(R.xml.pref_audio);
    }
}

File: app/src/main/java/code/name/monkey/retromusic/ui/fragments/settings/AudioSettings.kt
package code.name.monkey.retromusic.ui.fragments.settings

import android.content.Intent
import android.content.pm.PackageManager
import android.content.pm.ResolveInfo
import android.media.audiofx.AudioEffect
import android.os.Bundle
import androidx.preference.Preference

import code.name.monkey.retromusic.R
import code.name.monkey.retromusic.util.NavigationUtil
import code.name.monkey.retromusic.util.PreferenceUtil

/**
 * @author Hemanth S (h4h13).
 */

class AudioSettings : AbsSettingsFragment() {
    override fun invalidateSettings() {
        val findPreference = findPreference("equalizer")
        if (!hasEqualizer() && PreferenceUtil.getInstance().selectedEqualizer != "retro") {
            findPreference.isEnabled = false
            findPreference.summary = resources.getString(R.string.no_equalizer)
        } else {
            findPreference.isEnabled = true
        }
        findPreference.setOnPreferenceClickListener {
            NavigationUtil.openEqualizer(activity!!)
            true
        }


    }

    private fun hasEqualizer(): Boolean {
        val effects = Intent(AudioEffect.ACTION_DISPLAY_AUDIO_EFFECT_CONTROL_PANEL)

        val pm = activity!!.packageManager
        val ri = pm.resolveActivity(effects, 0)
        return ri != null
    }

    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        addPreferencesFromResource(R.xml.pref_audio)
    }
}
